File: Ambyint Assignment/ReadFileContinuously.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net.Sockets;
using System.IO;
using System.Net;

namespace ReadFileContinuously
{
    class Program
    {
        static void Main(string[] args)
        {
            // read the data file and write to the output every 150 ms until the user hits Enter key
            while (!(Console.KeyAvailable && Console.ReadKey(true).Key == ConsoleKey.Enter))
            {
                DateTime startDateTime = DateTime.Now;
                DateTime endDateTime = DateTime.Now;
                int counter = 0;
                string line = "";
                int sum = 0;

                System.IO.StreamReader file = new System.IO.StreamReader("data");
                System.IO.StreamWriter output = new System.IO.StreamWriter("output");

                // Continuously read the file line by line.  
                while ((line = file.ReadLine()) != null && (endDateTime - startDateTime).Milliseconds < 150)
                {
                    Int16 value = Convert.ToInt16(line);
                    sum += value;

                    counter++;
                    endDateTime = DateTime.Now;
                }

                output.WriteLine("sum of {0} lines read in {1} milliseconds = {2}", counter, (endDateTime - startDateTime).Milliseconds, sum);
                output.Close();
                output.Dispose();

                // Rename the output.txt file.
                if (File.Exists("outputOld"))
                {
                    File.Delete("outputOld");
                }

                File.Move("output", "outputOld");

                file.Close();
                file.Dispose();

                System.Console.WriteLine("sum of {0} lines read in {1} milliseconds = {2}", counter, (endDateTime - startDateTime).Milliseconds, sum);
                sum = 0;
            }
        }
    }
}

File: ReadFileContinuously.cpp
// ReadFileContinuously.cpp : Defines the entry point for the console application.
//
#include "stdafx.h"
#include <chrono>
#include <string>
#include <fstream>
#include <iostream>
#include <iomanip>
#include <io.h>
#include <Windows.h>
#include <stdio.h>

using namespace std;

int main()
{
	// read the data file and write to the output every 150 ms until the user hits Enter key
	bool exit = false;

	cout << "Hit the <ENTER> key to stop.\n";

	while (exit == false)
	{
		if (GetAsyncKeyState(VK_RETURN))
		{
			exit = true;
		}

		auto begin = chrono::high_resolution_clock::now();
		auto end = chrono::high_resolution_clock::now();
		auto dur = end - begin;
		auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(dur).count();

		int counter = 0, sum = 0;

		std::ifstream data;
		data.open("data");

		// Continuously read the file line by line.  
		while (data.is_open() && !data.eof() && ms < 150)
		{
			std::string line = "";
			std::getline(data, line);
			int value = atoi(line.c_str());
			sum += value;

			counter++;
			end = chrono::high_resolution_clock::now();
			dur = end - begin;
			ms = std::chrono::duration_cast<std::chrono::milliseconds>(dur).count();
		}

		std::ofstream output, outputOld;
		output.open("output");
		output << "sum of " << counter << " lines read in " << ms << " milliseconds = " << sum;
		output.close();

		std::cout << "sum of " << counter << " lines read in " << ms << " milliseconds = " << sum << "\n";
		sum = 0;
	}

    return 0;
}

